docs/backend.md

```diff
@@ -7,3 +7,78 @@ The IDP is a Rails application, that follows many typical Rails conventions.
 For consistency, we use [Faraday](https://github.com/lostisland/faraday)
 when making HTTP requests. We also wire in notifications so we can
 [log metrics on these requests](../config/initializers/faraday.rb)
+
+
+## Forms, FormResponse, Analytics, and Controllers
+
+We aim to keep Controllers simple and lean, and put business logic in Form
+classes, and hand those results (FormResponse) to our Analytics class to get
+logged in a consistent way.
+
+### FormResponse
+
+The [FormResponse](../app/services/form_response.rb) is a simple structure to help
+bundle up properties for logging. **Do not put PII or sensitive information
+inside these** because they are intended to be logged.
+
+```ruby
+FormResponse.new(
+  success: true | false,
+  errors: Hash | ActiveModel::Errors,
+  extra: Hash,
+)
+```
+
+### Forms
+
+We use `ActiveModel::Model` validations to help build useful error structures.
+
+Forms should have a `#submit` method that returns a `FormResponse`.
+- `success:` is usually `#valid?` from ActiveModel
+- `errors:` is usually `errors` from ActiveModel
+- `extra:` is, by convention, a method called `extra_analytics_attributes` that
+  returns a Hash
```

Comment:
Maybe it gets too much into the weeds, but I could see the `pii_like_keypaths` value getting some raised eyebrows, and could be something we'd want to document somewhere.